apps/src/music/analytics/AnalyticsReporter.ts

```diff
@@ -101,6 +101,11 @@ export default class AnalyticsReporter {
   async initialize(): Promise<void> {
     const response = await fetch(API_KEY_ENDPOINT);
     const responseJson = await response.json();
+
+    if (!responseJson.key) {
+      return;
```

Comment:
Do I possibly need to return a resolved promise?

Reply:
Might be good to `throw` an error here actually, so that we can catch and handle it ourselves in `startSession()`. In which case, we wouldn't set `this.sessionInProgress` to true, so all future analytics calls will just get blocked (which is what we want).

We could even go so far as to fully disable Amplitude reporting in development? Looks like we do that in our other more general Amplitude reporter ([here](https://github.com/code-dot-org/code-dot-org/blob/d2b9aa6cc3f85d8d6cab63409f93d28af1b6a047/apps/src/lib/util/AnalyticsReporter.js#L85)), but also include an ALWAYS_PUT flag that developers can set locally if they'd still like to test analytics (similar to what we do in [firehose.js](https://github.com/code-dot-org/code-dot-org/blob/d2b9aa6cc3f85d8d6cab63409f93d28af1b6a047/apps/src/lib/util/firehose.js#L71) as well)

Reply:
> Might be good to `throw` an error here actually, so that we can catch and handle it ourselves in `startSession()`. In which case, we wouldn't set `this.sessionInProgress` to true, so all future analytics calls will just get blocked (which is what we want).

This felt like a good solution for now, thanks!